### DISCUSSION TITLE
postgis 3.4 PG Debian 13.14-1.pgdg110+2 psql version 16 by default instead of version 13

### DISCUSSION THREAD

**Discussion by linnashan06** [postgis/docker-postgis#381] [Q&A] [answered]
Hi,
quite surprised when I see : 

```
postgres@288b4bdc8a93:~$ psql
psql (16.2 (Debian 16.2-1.pgdg110+2), server 13.14 (Debian 13.14-1.pgdg110+2))
Type "help" for help.

postgres=#
```

The image is based on PG13.14v, but the default clien psql  runs  at v16, there is any reason to make like this?
Moreover, all client versions were installed, this is not case for the image postgres:13.14-bullseye

```
root@288b4bdc8a93:/# find / -name psql
/usr/bin/psql
/usr/lib/postgresql/13/bin/psql
/usr/lib/postgresql/10/bin/psql
/usr/lib/postgresql/11/bin/psql
/usr/lib/postgresql/12/bin/psql
/usr/lib/postgresql/14/bin/psql
/usr/lib/postgresql/15/bin/psql
/usr/lib/postgresql/16/bin/psql
root@288b4bdc8a93:/#
root@288b4bdc8a93:/# su - postgres
```
By default, pg_config points to v16.
```
postgres@288b4bdc8a93:~$ pg_config
BINDIR = /usr/lib/postgresql/16/bin
DOCDIR = /usr/share/doc/postgresql-doc-16
HTMLDIR = /usr/share/doc/postgresql-doc-16
INCLUDEDIR = /usr/include/postgresql
PKGINCLUDEDIR = /usr/include/postgresql
INCLUDEDIR-SERVER = /usr/include/postgresql/16/server
LIBDIR = /usr/lib/x86_64-linux-gnu
PKGLIBDIR = /usr/lib/postgresql/16/lib
LOCALEDIR = /usr/share/locale
MANDIR = /usr/share/postgresql/16/man
SHAREDIR = /usr/share/postgresql/16
SYSCONFDIR = /etc/postgresql-common
```
I wish the client version fixed at 13 when I build the image, how can I achieve that ?
Thank you for your reply.

**Comment by ImreSamu** [comment]
> The image is based on PG13.14v, but the default clien psql runs at v16, there is any reason to make like this?
> I wish the client version fixed at 13 when I build the image, how can I achieve that ?

I actually don't know the answer..
( But it's also possible that I don't understand the question... because you didn't provide the minimal Dockerfile. )

- The `postgis/postgis:13-3.4` client is `psql (PostgreSQL) 13.14 (Debian 13.14-1.pgdg110+2)`   [1]
   - Dockerfile : https://github.com/postgis/docker-postgis/blob/master/16-3.4/Dockerfile 
- and many things can install a psql client (e.g., conda) 
    - you need to check step by step what installs the new psql client
- perhaps you might find more help on [StackOverflow.](https://stackoverflow.com/)

[1]  

`postgis/postgis:13-3.4`

```bash
$ docker run -it --rm   postgis/postgis:13-3.4 bash -c "psql --version && which psql && find / -name psql "
psql (PostgreSQL) 13.14 (Debian 13.14-1.pgdg110+2)
/usr/bin/psql
/usr/lib/postgresql/13/bin/psql
/usr/bin/psql
```

`postgres:13-bullseye`

```bash
$ docker run -it --rm   postgres:13-bullseye bash -c "psql --version && which psql && find / -name psql "
psql (PostgreSQL) 13.14 (Debian 13.14-1.pgdg110+2)
/usr/bin/psql
/usr/lib/postgresql/13/bin/psql
/usr/bin/psql
```

`postgis/postgis:13-3.4-alpine`
```bash
$ docker run -it --rm   postgis/postgis:13-3.4-alpine bash -c "psql --version && which psql && find / -name psql "
psql (PostgreSQL) 13.14
/usr/local/bin/psql
/usr/local/bin/psql
```

**Reply by linnashan06** [reply, marked as answer]
Thank you for your reply, you're totally right ! My team worker tried to modify the Dockerfile for adding the oracle_fdw extension. He coded ```apk add postgresql-server-dev-all```, which should be ```postgresql-server-dev-13```
